### PR TITLE
[ALM] flaky alm tests

### PR DESCRIPTION
**Pull request by eustlb**
# What does this PR do?

two things:
1. fix the flakyness of test_mismatching_num_audio_tokens
it duplicates audio[:1, ...] and expects the model to raise ValueError because total audio tokens no longer match total placeholders. This only fires if the duplicated sample contributes ≥ 1 audio token. For encoders with aggressive temporal downsampling like glmasr,  the audio mask can result to 0 audio tokens,  making that duplicating this remains 0 tokens and therefore no error is raised → flaky test

2. remove overwrittes of `create_audio_mask`

### CODE REVIEW

File: tests/alm_tester.py [hunk 1]
```diff
@@ -98,11 +99,15 @@ def create_audio_mask(self):
...
+        At least one batch index is pinned to a full-length mask.
         """
+        # Use a locally-seeded RNG so repeated calls within a test produce the same mask
+        rng = random.Random(0)
         # Sample lengths in [1, feat_seq_length] and offsets in [0, feat_seq_length - length]
-        lengths = ids_tensor([self.batch_size], vocab_size=self.feat_seq_length).abs() + 1
+        lengths = ids_tensor([self.batch_size], vocab_size=self.feat_seq_length, rng=rng).abs() + 1
         lengths = lengths.clamp(max=self.feat_seq_length)
-        offsets = ids_tensor([self.batch_size], vocab_size=self.feat_seq_length).abs()
+        lengths[rng.randint(0, self.batch_size - 1)] = self.feat_seq_length
```
Comment (vasqu) [inline tests/alm_tester.py:112]: I would highlight this a bit - I assume this is the new forcing of at least one valid entry

File: tests/alm_tester.py [hunk 2]
```diff
@@ -158,6 +163,19 @@ class ALMModelTest(MultiModalModelTest):
...
+        # Therefore we substitute a full-ones mask instead.
+        original_create_audio_mask = self.model_tester.create_audio_mask
+        self.model_tester.create_audio_mask = lambda: torch.ones(
+            [self.model_tester.batch_size, self.model_tester.feat_seq_length], dtype=torch.bool, device=torch_device
+        )
+        try:
+            super().test_sdpa_can_dispatch_on_flash()
+        finally:
+            self.model_tester.create_audio_mask = original_create_audio_mask
+
```
Comment (vasqu) [inline tests/alm_tester.py]: Can we not just use mock patching, e.g. with ...? I think it's a bit more elegant than manually doing this (and safer because these are written for exactly this)

File: tests/alm_tester.py [hunk 3]
```diff
@@ -168,6 +186,12 @@ def test_mismatching_num_audio_tokens(self):
         audio_feature_key = self.model_tester.get_audio_feature_key()
         audio_mask_key = self.model_tester.audio_mask_key
 
+        # Pick the batch index `create_audio_mask` pinned to full length — guaranteed to
+        # contribute > 0 audio tokens even for encoders that aggressively downsample
+        # (e.g. GlmAsr), so duplicating it in Test 2 reliably moves the audio-token total.
+        audio_token_id = self.model_tester.audio_token_id
+        dup_idx = int((input_dict["input_ids"] == audio_token_id).sum(-1).argmax().item())
```
Comment (vasqu) [inline tests/alm_tester.py:197]:
```suggestion
        dup_idx = int((input_dict[audio_feature_key] == audio_token_id).sum(-1).argmax().item())
```
just for consistency?

Comment (vasqu) [reply tests/alm_tester.py:197]: Super nit: And maybe `dup_idx` --> `audio_token_idx`. I think that's a bit more clearer

**Review by vasqu — APPROVED** [review]
Honestly a lot of nitpicking, I would only really change the sdpa on flash dispatch test to use mock patching instead of manually patching

File: tests/alm_tester.py (hunk 1 above)
Comment (eustlb) [reply tests/alm_tester.py:112]: agree, important to hightlight! adding a comment

File: tests/alm_tester.py (hunk 2 above)
Comment (eustlb) [reply tests/alm_tester.py]: absolutely! was not aware of that, thanks for the suggestion 🙏
